src/main/kotlin/pt/up/fe/ni/website/backend/controller/AuthController.kt

```diff
@@ -28,10 +31,29 @@ class AuthController(val authService: AuthService) {
         return mapOf("access_token" to accessToken)
     }
 
+    @PreAuthorize("isAuthenticated()")
     @GetMapping
-    @PreAuthorize("hasRole('MEMBER')")
-    fun checkAuthentication(): Map<String, Account> {
+    fun checkAuthentication(): Map<String, Any> {
+        val authentication = SecurityContextHolder.getContext().authentication
         val account = authService.getAuthenticatedAccount()
-        return mapOf("authenticated_user" to account)
+        return mapOf(
+            "authenticated_user" to account,
+            "jwt_permissions" to authentication.authorities.map { it.toString() }.toList()
+        )
+    }
+
+    @PreAuthorize("@authService.hasPermission(#permission.trim().toUpperCase())")
```

Comment:
I think it'd be better to do `trim().toUpperCase()` in the service so we don't write it in every annotation. Wdyt?